File: 程式語言 V2答案檔/C#/609.cs
using System;

public class TQC609
{
    public static void Main(string[] args)
    {
        int[] monthDays = {31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
        string[] inputs = Console.ReadLine().Split(' ');
        int year = Convert.ToInt32(inputs[0]);
        int month = Convert.ToInt32(inputs[1]);
        int day = Convert.ToInt32(inputs[2]);
        int answer = 0;

        if (IsInputIllegal(year, month, day)) { // input is illegal, print error
            Console.WriteLine("error");
        }
        else {
            for (int i = 0; i < month - 1; i++) {
                answer += monthDays[i];
            }
            answer += day + (IsLeapYear(year) && month >= 3 ? 1 : 0);
            Console.WriteLine(answer);
        }
    }

    // check the input is illegal or not
    public static bool IsInputIllegal(int year, int month, int day) {
        int[] monthDays = {31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
        if (IsLeapYear(year)) {
            monthDays[1] += 1;
        }
        
        if (day > monthDays[month - 1])
            return true;
        else
            return false;
    }

    public static bool IsLeapYear(int year) {
        if (year % 400 == 0)
            return true;
        else if ((year % 4 == 0) && (year % 100 != 0))
            return true;
        else
            return false;
    }
}

File: 程式語言 V2答案檔/PHP/609.php
<?php
    $array = array(31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31);
    
    $data = array_map('intval', explode(' ', fgets(STDIN)));
    $year = $data[0];
    $month = $data[1];
    $day = $data[2];
    
    if($year % 400 == 0 || ($year % 100 != 0 && $year % 4 == 0)){
        $array[1] = 29;
    }

    if($month > 12 || $month < 1){
        printf("error\n");
        return;
    }

    if($day < 0 || $day > $array[$month-1]){
        printf("error\n");
        return;
    }

    printf("%d\n", array_sum(array_slice($array, 0, $month-1)) + $day);
?>